packages/app/src/lib/components/ChatArea.svelte

```diff
@@ -76,9 +84,58 @@
       isSequentialMovedAnnouncement;
     return mergeWithPrevious;
   }
+
+  $effect(() => {
+    if (!messagesLoaded || !virtualizer || !globalState.channel) return;
+
+    // Only restore scroll position if we've switched to a different channel
+    if (globalState.channel.id !== currentChannelId) {
+      currentChannelId = globalState.channel.id;
+      
+      // Restore scroll position if available
+      const savedPosition = globalState.getScrollPosition(globalState.channel.id);
+      console.log("Restoring scroll position for channel", globalState.channel.id, "position:", savedPosition);
+      
+      if (!isRestoringScroll) {
+        isRestoringScroll = true;
+        // Scroll to the saved position in the next frame
+        requestAnimationFrame(() => {
+          virtualizer.scrollToIndex(savedPosition, { align: "start" });
+          isRestoringScroll = false;
+        });
+      }
+    }
+  });
+
+  // Handle scroll events to save position and check if at bottom
+  function handleScroll(e: UIEvent) {
+    if (!viewport) return;
+    
+    const { scrollTop, scrollHeight, clientHeight } = viewport;
+    const isNearBottom = scrollHeight - scrollTop - clientHeight < 100; // 100px threshold
+    if (isNearBottom !== isAtBottom) {
+      isAtBottom = isNearBottom;
+    }
+    
+    // Save scroll position
+    const currentIndex = Math.floor(scrollTop / 50);
+    if (globalState.channel) {
+      globalState.saveScrollPosition(globalState.channel.id, currentIndex);
+    }
```

Comment:
This seems to cause a lot of drift for me as I switch back and forth between channels until it finds a sweet spot. Why do we divide by 50?  Because it's half of 100? Maybe the "magic" numbers could be defined as descriptive consts somewhere as well. I tend to like that even if it's verbose just because it helps with readability for others.

Reply:
i added a function which should clean this up and doesn't use magic numbers